src/dlangui/platforms/sdl/sdlapp.d

```diff
@@ -263,6 +263,73 @@ class SDLWindow : Window {
         Log.d("SDLWindow.close()");
         _platform.closeWindow(this);
     }
+    
+    override bool setWindowState(WindowState newState, bool activate = false, Rect newWindowRect = RECT_VALUE_IS_NOT_SET) {
+        // override for particular platforms
+        
+        if (_win is null)
+            return false;
+        
+        bool res = false;
+        
+        // change state
+        switch(newState) {
+            case WindowState.maximized:
+                if (_windowState != WindowState.maximized)
+                    SDL_MaximizeWindow(_win);
+                res = true;
+                break;
+            case WindowState.minimized:
+                if (_windowState != WindowState.minimized)
+                    SDL_MinimizeWindow(_win);
+                res = true;
+                break;
+            case WindowState.hidden:
+                if (_windowState != WindowState.hidden)
+                    SDL_HideWindow(_win);
+                res = true;
+                break;
+            case WindowState.normal:
+                if (_windowState != WindowState.normal) {
+                    SDL_RestoreWindow(_win);
+                }
+                res = true;
+                break;
+            default:
+                break;
+        }
+        // change size and/or position
+        if (newWindowRect != RECT_VALUE_IS_NOT_SET && (newState == WindowState.normal || newState == WindowState.unspecified)) {
+               
+            if (newWindowRect.top == int.min || newWindowRect.left == int.min) {
```

Comment:
You're welcome!)) I like review for the ability to write code concentrating on solving the problem and after polish it to improve readability etc using remarks from others.